databricks_cli/stack/api.py

```diff
@@ -0,0 +1,336 @@
+        of the physical job_id of the job on databricks. deploy_output is the output of the job
+        from databricks when a GET request is called for it.
+        """
+        job_settings = resource_properties  # resource_properties of jobs are solely job settings.
+
+        if physical_id:
+            job_id = physical_id['job_id']
+            self.update_job(job_settings, physical_id['job_id'])
+        else:
+            job_id = self.put_job(job_settings)
+        click.echo("Job deployed on Databricks with job_id %s" % job_id)
+        physical_id = {'job_id': job_id}
+        deploy_output = self.jobs_client.get_job(job_id)
+        return physical_id, deploy_output
+
+    def deploy_resource(self, resource_config, resource_status=None):  # overwrite to be added
+        """
+        Deploys a resource given a resource information extracted from the stack JSON configuration
+        template.
+
+        :param resource_config: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE
+        and RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :param resource_status: A dict of the resource's deployment info from the last
+        deployment. Will be None if this is the first deployment.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {...}}
+        :return: dict resource_status- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        resource_id = resource_config[RESOURCE_ID]
+        resource_service = resource_config[RESOURCE_SERVICE]
+        resource_properties = resource_config[RESOURCE_PROPERTIES]
+        physical_id = resource_status[RESOURCE_PHYSICAL_ID] if resource_status else None
+
+        if resource_service == JOBS_SERVICE:
+            click.echo("Deploying job '%s' with properties: \n%s \n" % (resource_id, json.dumps(
+                resource_properties, indent=2, separators=(',', ': '))), nl=False)
+            physical_id, deploy_output = self.deploy_job(resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource service '%s' not supported" % resource_service)
+
+        new_resource_status = {RESOURCE_ID: resource_id, RESOURCE_SERVICE: resource_service,
+                               RESOURCE_DEPLOY_TIMESTAMP: datetime.now(),
+                               RESOURCE_PHYSICAL_ID: physical_id,
+                               RESOURCE_DEPLOY_OUTPUT: deploy_output}
+        return new_resource_status
+
+    def validate_status(self, stack_status):
```

Comment:
Since the status file is a generated file, we need to better advise people how to handle the error. Consider giving better description in error message.

Reply:
sure thing